src/config-loader.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Config handling - here because of strange babel errors.
+ * By Mia
+ * @author mia-pi-git
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+// @ts-ignore no typedef file
+import * as defaults from '../config/config-example';
+
+export type Configuration = typeof defaults;
+
+export function load(invalidate = false): Configuration {
+	const configPath = path.resolve(__dirname, '../../config/config.js');
+	if (invalidate) delete require.cache[configPath];
+	let config = defaults;
+	try {
+		config = {...config, ...require(configPath)};
+	} catch (err: any) {
+		if (err.code !== 'MODULE_NOT_FOUND') throw err; // Should never happen
+
+		console.log("config.js doesn't exist - creating one with default settings...");
+		fs.writeFileSync(
+			configPath,
+			fs.readFileSync(path.resolve(__dirname, '../../config/config-example.js'))
+		);
+	}
+	// so it's loaded after actions is originally loaded (preventing a crash)
+	process.nextTick(() => {
+		if (config.actions) {
+			Object.assign(require('./actions').actions, config.actions);
+		}
+	});
+	return config;
+}
+export const Config: Configuration = load();
+
+if (Config.watchconfig) {
```

Comment:
I really do not like file watchers, and I especially do not like them in production.
- The atomicity of writing a file depends on the editor; there is no guarantee that the configuration file will be in a well-defined state when you load it.
- I should be able to save the file--something I do compulsively while editing--without affecting the service
- Loading the configuration should not be an asynchronous task. How do you report errors to the user?

I also don't like updating the configuration while the process is currently running. How do you know the service will be in a consistent state afterwards?

What I would recommend is not bothering with this at all. Just restart the process when you want to change the configuration. Your restart should be zero-downtime anyway.

Reply:
I would like to keep this for development purposes. We can just disable it in production.

Reply:
OK, no objection from me as long as you find it useful in dev.